cmd/controller/main.go

```diff
@@ -65,7 +72,18 @@ func main() {
 	if err := images.Validate(); err != nil {
 		log.Fatal(err)
 	}
-	sharedmain.MainWithContext(injection.WithNamespaceScope(signals.NewContext(), *namespace), ControllerLogKey,
+	controller.DefaultThreadsPerController = *threadsPerController
+
+	cfg := sharedmain.ParseAndGetConfigOrDie()
+	cfg.QPS = float32(*qps)
+	cfg.Burst = *burst
+
+	ctx := injection.WithNamespaceScope(signals.NewContext(), *namespace)
+	if !*enableLeaderElection {
```

Comment:
The leader election part worries me a bit. I think we'd want to setup some end to end testing of this before making it available. 

Do you think it's worth trying to include in this change, or separating out to another PR where we can figure out the necessary testing plan?

Reply:
To my understanding, I _think_ if we remove the option of enabling leader election, it might be a breaking change, as the current release seems to support it via https://github.com/tektoncd/pipeline/blob/master/config/config-leader-election.yaml. I am not too super familiar with knative but looking through PR https://github.com/knative/pkg/pull/1476, looks like one could enable it by setting `EnabledComponents`.

I could break that out in a separate commit though, if we want to add some sort end to end test for it. I might need some guidance with that though, not sure what we might test in addition to what knative is already testing for :)

Either way, I guess we will want to add something in the release notes, if folks are using leader election now and how to use going forward.

Reply:
Ah, maybe my mistake then. It's not actually clear to me if that works. The comment is "an inactive but valid configuration follows": https://github.com/tektoncd/pipeline/blob/master/config/config-leader-election.yaml#L24

But either way - I'm not quite sure how not including this flag would be a breaking change. It didn't exist before, why do we need it here?